R/processes.R

```diff
@@ -61,16 +61,38 @@ create_processes <- function(
       )
     )
   }
+  
+  # =====================================================
+  # Competing Hazard Outcomes (Infections and Recoveries)
+  # =====================================================
+  
+  infection_outcome <- CompetingOutcome$new(
+    targeted_process = function(timestep, target){
+      infection_process_resolved_hazard(timestep, target, infection_outcome$source_humans,
+                                        variables, renderer, parameters, prob = rate_to_prob(infection_outcome$rates))
+    },
+    size = parameters$human_population
+  )
+  
+  recovery_outcome <- CompetingOutcome$new(
+    targeted_process = function(timestep, target){
+      recovery_process_resolved_hazard(timestep, target, variables, parameters, renderer)
+    },
+    size = parameters$human_population
+  )
+  
+  create_infection_recovery_hazard_process <- CompetingHazard$new(
+    outcomes = list(infection_outcome, recovery_outcome)
+  )
 
   # ==============================
   # Biting and mortality processes
   # ==============================
-  # schedule infections for humans and set last_boosted_*
+  # simulate infections for humans and set last_boosted_*
   # move mosquitoes into incubating state
   # kill mosquitoes caught in vector control
-  processes <- c(
-    processes,
-    create_biting_process(
+  create_infection_rates_process <- function(timestep){
+    biting_process(
```

Comment:
couldn't we keep the function factory in biting_process.R?

In this file we've been calling function factories from *_process.R files. But now we're making the functions in processes.R. It feels a bit inconsistent. And verbose since we have a lot of processes to create.